### PR TITLE
Update geofence domain models to location module

### PR DESCRIPTION
**Pull request by evan-masseau**
# Description

Updated domain models for geofencing:
- KlaviyoGeofence: Core geofence data model with JSON serialization
- KlaviyoGeofenceTransition: Enter/exit event types with conversion from system events
- Comprehensive test coverage including JSON round-trip and edge cases

These models provide some foundation for geofence fetching PR coming next.

## Due Diligence

- [ ] I have tested this on an emulator and/or a physical device.
- [x] I have added sufficient unit/integration tests of my changes.
- [ ] I have adjusted or added new test cases to team test docs, if applicable.
- [ ] I am confident these changes are compatible with all Android versions the SDK currently supports.

## Release/Versioning Considerations

- [ ] `Patch` Contains internal changes or backwards-compatible bug fixes.
- [ ] `Minor` Contains changes to the public API. 
- [ ] `Major` Contains **breaking** changes.
- [ ] Contains readme or migration guide changes.
  - If so, please merge to a feature branch so documentation updates only go live upon official release.
- [ ] This is planned work for an upcoming release.
  - If no, author or reviewer should account for this in a release plan, or describe why not below.

## Related Issues/Tickets
 
 
 

---

> [!NOTE]
> Adds location geofence models with composite IDs and updates analytics parsing to include companyId, with comprehensive tests.
> 
> - **Location**:
>   - **KlaviyoGeofence**: new core model using composite ID `companyId:locationId`, with ID validation, JSON constants, and `toJson()`; adds converters from `FetchedGeofence`, `Geofence`, and `JSONObject`.
>   - **KlaviyoGeofenceTransition**: new enum and `GeofencingEvent` converter.
>   - Minor cleanup in `KlaviyoLocationManager` to use new extensions.
> - **Analytics**:
>   - **FetchedGeofence**: add `companyId` field; update JSON parsers to accept `companyId` and construct via `Registry.config.apiKey` in `FetchGeofencesRequest`.
> - **Tests**:
>   - Add tests for `KlaviyoGeofence` (JSON round-trip, parsing, validation) and `KlaviyoGeofenceTransition`.
>   - Update analytics tests to reflect `companyId` in `FetchedGeofence` and array parsing.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit a103dffe04808145d22c1367e6e6af4891e5c3bd. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: sdk/location/src/main/java/com/klaviyo/location/KlaviyoGeofence.kt [hunk 1]
```diff
@@ -73,70 +42,56 @@ data class KlaviyoGeofence(
...
-        put("radius", geofence.radius)
-    }.toString()
+    fun toJson() = JSONObject().apply {
+        put(KEY_ID, id)
+        put(KEY_LATITUDE, latitude)
+        put(KEY_LONGITUDE, longitude)
+        put(KEY_RADIUS, radius)
+    }
 
     companion object {
```
Comment (dan-peluso) [inline sdk/location/src/main/java/com/klaviyo/location/KlaviyoGeofence.kt]: we don't need to add a @Keep on this do we?

Comment (evan-masseau) [reply sdk/location/src/main/java/com/klaviyo/location/KlaviyoGeofence.kt]: haven't needed it for any other constants. this should be ok since it has a clear reference

File: sdk/location/src/main/java/com/klaviyo/location/KlaviyoGeofence.kt [hunk 2]
```diff
@@ -73,70 +42,56 @@ data class KlaviyoGeofence(
...
-            longitude = geofence.longitude,
-            radius = geofence.radius
-        )
+/**
+ * Create a [KlaviyoGeofence] from a [FetchedGeofence] API response object.
+ *
+ * Note: this is where we combine the company ID and location ID to form the geofence ID.
+ */
+fun FetchedGeofence.toKlaviyoGeofence(): KlaviyoGeofence = KlaviyoGeofence(
+    id = "${Registry.config.apiKey}-$id",
```
[the remark this answers is not in the document]
Comment (evan-masseau) [reply sdk/location/src/main/java/com/klaviyo/location/KlaviyoGeofence.kt]: No, but now that you mention it there's a cleaner way -- since FetchedGeofence comes from an API request, I should have the company ID in scope when building this data object, so i'll capture it as a field of `FetchedGeofence` instead of getting it from config.

File: sdk/location/src/main/java/com/klaviyo/location/KlaviyoGeofence.kt [hunk 3]
```diff
@@ -60,83 +29,93 @@ data class KlaviyoGeofence(
...
+                    "Invalid geofence ID format: '$id' - companyId must be exactly 6 characters, got ${extractedCompanyId.length}"
+                )
+            }
+            if (extractedLocationId.isEmpty()) {
+                Registry.log.warning(
+                    "Invalid geofence ID format: '$id' - locationId cannot be empty"
+                )
+            }
+        }
+    }
```
[the remark this answers is not in the document]
Comment (evan-masseau) [reply sdk/location/src/main/java/com/klaviyo/location/KlaviyoGeofence.kt]: registry.log does not throw.

File: sdk/location/src/main/java/com/klaviyo/location/KlaviyoGeofence.kt [hunk 4]
```diff
@@ -60,83 +29,93 @@ data class KlaviyoGeofence(
...
      * Company ID to which this geofence belongs, extracted from the geofence ID.
      */
-    val companyId: String = id.split('-').firstOrNull() ?: ""
+    val companyId: String = id.substringBefore(':')
 
     /**
      * Location ID to which this geofence belongs, extracted from the geofence ID.
      */
-    val locationId: String = id.split('-').getOrNull(1) ?: ""
+    val locationId: String = id.substringAfter(':', "")
```
[the remark this answers is not in the document]
Comment (evan-masseau) [reply sdk/location/src/main/java/com/klaviyo/location/KlaviyoGeofence.kt]: Ohh i see what you're saying now. Well because `id` is immutable, I don't think this needs to be computed with a `get()`. I mean its a micro-level difference, but we only need to do it once.

The data class evaluation runs top to bottom. So init runs first, logs the warnings, and then this sets `companyId` and `locationId`
